File: Models/Stics_soil_temperature/src/java/Stics_soil_temperature/Canopy_temp_avg.java
import  java.io.*;
import  java.util.*;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.time.LocalDateTime;
public class Canopy_temp_avg
{
    
    public Canopy_temp_avg() { }
    public void  Calculate_Model(soil_tempState s, soil_tempState s1, soil_tempRate r, soil_tempAuxiliary a,  soil_tempExogenous ex)
    {
        //- Name: canopy_temp_avg -Version: 1.0, -Time step: 1
        //- Description:
    //            * Title: canopy_temp_avg model
    //            * Authors: None
    //            * Reference: doi:http://dx.doi.org/10.1016/j.agrformet.2014.05.002
    //            * Institution: INRAE
    //            * ExtendedDescription: None
    //            * ShortDescription: None
        //- inputs:
    //            * name: min_canopy_temp
    //                          ** description : current minimum temperature
    //                          ** inputtype : variable
    //                          ** variablecategory : exogenous
    //                          ** datatype : DOUBLE
    //                          ** max : 50.0
    //                          ** min : -50.0
    //                          ** default : 0.0
    //                          ** unit : degC
    //            * name: max_canopy_temp
    //                          ** description : current maximum temperature
    //                          ** inputtype : variable
    //                          ** variablecategory : exogenous
    //                          ** datatype : DOUBLE
    //                          ** max : 50.0
    //                          ** min : -50.0
    //                          ** default : 0.0
    //                          ** unit : degC
        //- outputs:
    //            * name: canopy_temp_avg
    //                          ** description : current temperature amplitude
    //                          ** datatype : DOUBLE
    //                          ** variablecategory : state
    //                          ** max : 100.0
    //                          ** min : 0.0
    //                          ** unit : degC
        Double min_canopy_temp = ex.getmin_canopy_temp();
        Double max_canopy_temp = ex.getmax_canopy_temp();
        Double canopy_temp_avg;
        canopy_temp_avg = (max_canopy_temp + min_canopy_temp) / 2;
        s.setcanopy_temp_avg(canopy_temp_avg);
    }
}

File: Models/Stics_soil_temperature/src/r/Stics_soil_temperature/canopy_temp_avg.r
library(gsubfn)

#' canopy_temp_avg model
#'
#' This function compute the canopy_temp_avg model
#' @param min_canopy_temp (degC) current minimum temperature exogenous (0.0, -50.0-50.0) 
#' @param max_canopy_temp (degC) current maximum temperature exogenous (0.0, -50.0-50.0) 
#'
#' @return
#' \describe{
#'   \item{canopy_temp_avg (degC)}{current temperature amplitude state (0.0-50.0)} 
#' }
#' @export
model_canopy_temp_avg <- function (min_canopy_temp,
         max_canopy_temp){
    canopy_temp_avg <- 0.0
    canopy_temp_avg <- (max_canopy_temp + min_canopy_temp) / 2
    return (list('canopy_temp_avg' = canopy_temp_avg))
}